nca/NetworkConfig/NetworkConfigQuery.py

```diff
@@ -767,17 +767,22 @@ def exec(self, layer_name=None):
         peers_to_compare = self.config1.peer_container.get_all_peers_group()
         peers_to_compare |= self.disjoint_referenced_ip_blocks()
         captured_pods = self.config1.get_captured_pods(layer_name) | self.config2.get_captured_pods(layer_name)
+        different_conns_list = []
         for peer1 in peers_to_compare:
             for peer2 in peers_to_compare if peer1 in captured_pods else captured_pods:
                 if peer1 == peer2:
                     continue
                 conns1, _, _, _ = self.config1.allowed_connections(peer1, peer2, layer_name)
                 conns2, _, _, _ = self.config2.allowed_connections(peer1, peer2, layer_name)
                 if conns1 != conns2:
-                    explanation = f'Allowed connections from {peer1} to {peer2} are different\n' + \
-                                  conns1.print_diff(conns2, self.name1, self.name2)
-                    return QueryAnswer(False, self.name1 + ' and ' + self.name2 + ' are not semantically equivalent.',
-                                       explanation)
+                    different_conns_list.append(f'Allowed connections from {peer1} to {peer2}'
+                                                f' are different\n' + conns1.print_diff(conns2, self.name1, self.name2))
+        if different_conns_list:
+            explanation = different_conns_list[0] + '\n'
+            if self.output_config.printAllPairs:
+                explanation += '\n'.join(different_conns_list[1:])
+            return QueryAnswer(False, self.name1 + ' and ' + self.name2 + ' are not semantically equivalent.\n',
+                               explanation)
```

Comment:
I think that for efficiency considerations, it may be better to break from the loop if the output config only requires one counterexample. (relevant for all queries where this applies to).